jujugui/static/gui/src/app/components/inspector/inspector.js

```diff
@@ -72,19 +72,19 @@ class Inspector extends React.Component {
           title: service.get('name'),
           icon: service.get('icon'),
           component: <ServiceOverview
-            acl={this.props.acl}
-            addNotification={this.props.addNotification}
+            acl={nextProps.acl}
+            addNotification={nextProps.addNotification}
             changeState={changeState}
-            charm={this.props.charm}
-            clearState={this.props.clearState}
-            destroyService={this.props.destroyService}
-            displayPlans={this.props.displayPlans}
-            getUnitStatusCounts={this.props.getUnitStatusCounts}
-            modelUUID={this.props.modelUUID}
+            charm={nextProps.charm}
+            clearState={nextProps.clearState}
+            destroyService={nextProps.destroyService}
```

Comment:
This was generating the state using the previous props, instead of the new props.